hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -342,6 +353,35 @@ public SingleTransactionRecordBuilderImpl transactionID(@NonNull final Transacti
         return this;
     }
 
+    /**
+     * When we update nonce on the record, we need to update the body as well with the same transactionID.
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl syncBodyIdFromRecordId() {
+        final var newTransactionID = transactionID;
+        try {
+            final var signedTransaction = SignedTransaction.PROTOBUF.parseStrict(
+                    transaction.signedTransactionBytes().toReadableSequentialData());
+            final var existingTransactionBody =
+                    TransactionBody.PROTOBUF.parse(signedTransaction.bodyBytes().toReadableSequentialData());
+            final var body = existingTransactionBody
+                    .copyBuilder()
+                    .transactionID(newTransactionID)
+                    .build();
+            final var newBodyBytes = TransactionBody.PROTOBUF.toBytes(body);
+            final var newSignedTransaction =
+                    SignedTransaction.newBuilder().bodyBytes(newBodyBytes).build();
+            final var signedTransactionBytes = SignedTransaction.PROTOBUF.toBytes(newSignedTransaction);
```

Comment:
Do key checks still pass if we modify the signed bytes?

Reply:
I don't think it should. But some HapiTests are failing. Will look. Thanks!

Reply:
This isn't an issue because this transaction is a synthetic child transaction, not the bytes that the user signed.